REGRESSION (273978@main): JPEG2000 Images in this PDF fail to display

shallawa opened the pull request:
#### 4c3caa5b7d5d1bb40f73483c3c947edddb93871e
<pre>
REGRESSION (273978@main): JPEG2000 Images in this PDF fail to display
<a href="https://bugs.webkit.org/show_bug.cgi?id=271726">https://bugs.webkit.org/show_bug.cgi?id=271726</a>
<a href="https://rdar.apple.com/125291465">rdar://125291465</a>

Reviewed by Simon Fraser.

JPEG2000 can be supported only in PDF plugin, embedded objects and images. But
it should still be disabled anywhere else.

* LayoutTests/pdf/unifiedpdf/images/jpeg2000-embed-pdf-expected.html: Added.
* LayoutTests/pdf/unifiedpdf/images/jpeg2000-embed-pdf.html: Added.
* LayoutTests/pdf/unifiedpdf/resources/400x400-green-jpeg2000.pdf: Added.
* Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
(WebCore::allowableImageTypes):

Canonical link: <a href="https://commits.webkit.org/276734@main">https://commits.webkit.org/276734@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ac97ca76c152cb629f779be9443f3e1ba058938c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/45521 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/24647 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48060 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48189 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/41530 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/47828 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/28900 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22041 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/37312 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46099 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/21734 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39280 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18439 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19132 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40365 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/3567 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/41843 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/40692 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/49925 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/20509 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/17038 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/44388 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/21815 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/39432 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/43222 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10119 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22175 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/21503 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on LayoutTests/pdf/unifiedpdf/images/jpeg2000-embed-pdf.html:
```diff
@@ -0,0 +1,42 @@
+        if (window.testRunner)
+            testRunner.waitUntilDone();
+
+
```
Extra blank line.

smfr commented on LayoutTests/pdf/unifiedpdf/images/jpeg2000-embed-pdf.html:
```diff
@@ -0,0 +1,42 @@
+            testRunner.waitUntilDone();
+
+
+        addEventListener("load", async () => {
```
Don't need the `async`, there is no `await` here.

smfr commented on LayoutTests/pdf/unifiedpdf/images/jpeg2000-embed-pdf.html:
```diff
@@ -0,0 +1,42 @@
+            let rafs = 0;
+
+            let countRafs = (timestamp) => {
+                if (++rafs < 50)
```
If you enable UnifiedPDF via the header you should be able to avoid waiting for the PDF fade in animation.

shallawa replied:
Adding 
```
<!-- webkit-test-runner [ UnifiedPDFEnabled=true ] -->
```
did not prevent the PDF fade animation.

shallawa committed:
REGRESSION (273978@main): JPEG2000 Images in this PDF fail to display
https://bugs.webkit.org/show_bug.cgi?id=271726
rdar://125291465

Reviewed by Simon Fraser.

JPEG2000 can be supported only in PDF plugin, embedded objects and images. But
it should still be disabled anywhere else.

* LayoutTests/pdf/unifiedpdf/images/jpeg2000-embed-pdf-expected.html: Added.
* LayoutTests/pdf/unifiedpdf/images/jpeg2000-embed-pdf.html: Added.
* LayoutTests/pdf/unifiedpdf/resources/400x400-green-jpeg2000.pdf: Added.
* Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
(WebCore::allowableImageTypes):

Canonical link: https://commits.webkit.org/276734@main
